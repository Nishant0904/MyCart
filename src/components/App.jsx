import Header from './Header'
import PageContent from './PageContent'
import Footer from './Footer'
import { useState } from 'react'

const App = () => {
    const [cart, setCart] = useState([])

    return (
        <>
            <Header cart={cart} />
            <PageContent
                cart={cart}
                handleAddToCart={handleAddToCart}
                handleDeleteFromCart={handleDeleteFromCart}
            />
            <Footer />
        </>
    )

    function handleAddToCart(newProduct, quantity) {
        if (quantity > 0) {
            let productsToAdd = []
            for (let i = quantity; i > 0; i--) {
                productsToAdd.push(newProduct)
            }
            setCart([...cart, ...productsToAdd])
        }
    }

    function handleDeleteFromCart(currentProduct) {
        const cartWithoutProduct = cart.filter(
            (product) => product.title !== currentProduct.title,
        )
        setCart(cartWithoutProduct)
    }
}

export default App
