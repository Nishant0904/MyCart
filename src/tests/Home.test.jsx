import { describe, it, expect } from 'vitest'
import { render, screen, waitFor } from '@testing-library/react'
import App from '../components/App'
import { BrowserRouter } from 'react-router-dom'
import userEvent from '@testing-library/user-event'

describe('Home page', () => {
    it('Site first renders homepage', () => {
        render(<App />, { wrapper: BrowserRouter })
        expect(screen.getByTestId('home__hero-title')).toBeInTheDocument()
    })

    it('Home page matches snapshot', () => {
        render(<App />, { wrapper: BrowserRouter })
        expect(screen.getByTestId('home-section')).toMatchSnapshot()
    })

    it('Shop Collection button navigates to producst page', async () => {
        const user = userEvent.setup()
        render(<App />, { wrapper: BrowserRouter })

        const shopCollectionBtn = screen.getByRole('link', {
            name: 'Shop Collection',
        })
        expect(shopCollectionBtn.href).toContain('/products')
        await user.click(shopCollectionBtn)
        await waitFor(() => {
            expect(window.location.pathname).toBe('/products')
        })
    })
})
