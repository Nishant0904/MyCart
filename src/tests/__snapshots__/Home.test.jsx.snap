// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > Home page matches snapshot 1`] = `
<section
  class="sc-iBdnpw jfPBdp"
  data-testid="home-section"
>
  <h1
    data-testid="home__hero-title"
  >
    <span>
      Ever
    </span>
    ything You Could
    <br />
    <span>
      Ever
    </span>
     Ask For
  </h1>
  <a
    class="sc-gLLuof bgONbp"
    href="/products"
  >
    Shop Collection
  </a>
</section>
`;
